app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java

```diff
@@ -79,154 +102,169 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
-//                    @Override
-//                    public void onLocationResult(LocationResult locationResult) {
-//                        onLocationChanged(locationResult.getLastLocation());
-//                    }
-//                },
-//                Looper.myLooper());
-//    }
-//
-//    public void onLocationChanged(Location location) {
-//        // GPS may be turned off
-//        if (location == null) {
-//            return;
-//        }
-//
-//        mCurrentLocation = location;
-//        String msg = "Updated Location: " +
-//                Double.toString(location.getLatitude()) + "," +
-//                Double.toString(location.getLongitude());
-//        //Toast.makeText(this, msg, Toast.LENGTH_SHORT).show();
-//        displayLocation();
-//    }
-//
-//    public void onSaveInstanceState(Bundle savedInstanceState) {
-//        savedInstanceState.putParcelable(KEY_LOCATION, mCurrentLocation);
-//        super.onSaveInstanceState(savedInstanceState);
-//    }
-//
-//    @Override
-//    public void onMapLongClick(@NonNull LatLng latLng) {
-//
-//    }
+        map = googleMap;
+        getLocation();
+//        googleMap.addMarker(new MarkerOptions()
+//                .position(new LatLng(51.5072, 0.1276))
+//                .title("London"));
+//        LatLng latLng = new LatLng(51.5072, 0.1276);
+//        map.moveCamera(CameraUpdateFactory.newLatLng(latLng));
+//        map.animateCamera(CameraUpdateFactory.newLatLngZoom(latLng,15f));
+    }
+
+    private void getLocation() {
+        progressDialog.show();
+        if(Build.VERSION.SDK_INT>=23){
+            if(checkPermission()){
+                getDeviceLocation();
+            }
+            else{
+                requestPermission();
+            }
+        }
+        else{
+            getDeviceLocation();
+        }
+    }
+
+    private void requestPermission() {
+        ActivityCompat.requestPermissions(getActivity(),permissions_all,PERMISSION_CODE);
+    }
+
+    private boolean checkPermission() {
+        for(int i=0;i<permissions_all.length;i++){
+            int result= ContextCompat.checkSelfPermission(getContext(),permissions_all[i]);
+            if(result== PackageManager.PERMISSION_GRANTED){
+                continue;
+            }
+            else {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private void getDeviceLocation() {
+        locationManager=(LocationManager) getActivity().getSystemService(Service.LOCATION_SERVICE);
+        isGpsLocation=locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER);
+        isNetworklocation=locationManager.isProviderEnabled(LocationManager.NETWORK_PROVIDER);
+        if(!isGpsLocation && !isNetworklocation){
+            showSettingForLocation();
+            getLastlocation();
+        }
+        else{
+            getFinalLocation();
+        }
+    }
+
+    private void getLastlocation() {
+        if(locationManager!=null) {
```

Comment:
Something else you can do to make your code easier to understand is short circuiting. Here you can reduce nesting by writing this as:
if (locationManager == null) { return; }

and then its clear the rest of the method expects this to be true without relying on counting nests.